File: 251004/Yelkin/4-5/CommentService/src/main/java/by/yelkin/CommentService/entity/Comment.java
package by.yelkin.CommentService.entity;


import lombok.Getter;
import lombok.Setter;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;

import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

@Getter
@Setter
@Table("tbl_comment")
public class Comment {

    @PrimaryKeyColumn(name = "country", ordinal = 0, type = PrimaryKeyType.PARTITIONED)
    private String country;

    @PrimaryKeyColumn(name = "topic_id", ordinal = 1, type = PrimaryKeyType.CLUSTERED)
    private Long topicId;

    @PrimaryKeyColumn(name = "id", ordinal = 2, type = PrimaryKeyType.CLUSTERED)
    private Long id;

    @Column("content")
    private String content;
}

File: 251003/Kukhatskavolets/microblog/discussion/src/main/kotlin/by/kukhatskavolets/discussion/entities/Comment.kt
package by.kukhatskavolets.discussion.entities

import jakarta.validation.constraints.Size
import org.springframework.data.cassandra.core.mapping.PrimaryKey
import org.springframework.data.cassandra.core.mapping.Table

@Table("tbl_comment")
data class Comment(
    @PrimaryKey
    var key: CommentKey,

    val tweetId: Long,

    @field:Size(min = 2, max = 2048)
    val content: String  = ""
)